mediapipe/graphs/instantmotiontracking/calculators/gl_animation_overlay_calculator.cc

```diff
@@ -626,36 +635,45 @@ ::mediapipe::Status GlAnimationOverlayCalculator::GlSetup() {
 
     // vertex position in threespace
     attribute vec4 position;
+    attribute vec3 normal;
 
     // texture coordinate for each vertex in normalized texture space (0..1)
     attribute mediump vec4 texture_coordinate;
 
     // texture coordinate for fragment shader (will be interpolated)
-    varying mediump vec2 sample_coordinate;
+    varying mediump vec3 sample_coordinate;
+    varying mediump vec3 vNormal;
 
     void main() {
-      sample_coordinate = texture_coordinate.xy;
+      sample_coordinate = texture_coordinate.xyz;
       mat4 mvpMatrix = perspectiveMatrix * modelMatrix;
       gl_Position = mvpMatrix * position;
+
+      vec4 tmpNormal = mvpMatrix * vec4(normal, 1.0);
+      vNormal = normalize(tmpNormal.xyz);
     }
   )";
 
   const GLchar *frag_src = R"(
     precision mediump float;
 
-    varying vec2 sample_coordinate;  // texture coordinate (0..1)
+    varying vec3 sample_coordinate;  // texture coordinate (0..1)
+    varying vec3 vNormal;
     uniform sampler2D texture;  // texture to shade with
 
+    // Define ambient lighting constants
+    const vec3 ambient = vec3(-0.1, -0.1, -0.1);
```

Comment:
Is this our light direction?

Reply:
This is our ambient (unchanging) lighting for the scene that is applied to every object. The actual lighting (spotlight effects) and the use of normals to create shading on the object is being done on a separate branch and will be part of PR #5.

Reply:
What does the three values of 0.1 represent? Can you document them and explain how you decide those numbers? In general, constants beyond 0 and 1 would benefit from some documentation or use a named variable whose name is self explanatory.